quetz/main.py

```diff
@@ -916,19 +916,50 @@ def post_channel_member(
     new_member: rest_models.PostMember,
     channel: db_models.Channel = Depends(get_channel_or_fail),
     dao: Dao = Depends(get_dao),
+    db=Depends(get_db),
     auth: authorization.Rules = Depends(get_rules),
 ):
 
-    auth.assert_add_channel_member(channel.name, new_member.role)
+    if not dao.get_user_by_username(new_member.username):
+        raise HTTPException(
+            status_code=status.HTTP_404_NOT_FOUND,
+            detail=f"user {new_member.username} not found",
+        )
 
+    auth.assert_list_channel_members(channel.name)
     channel_member = dao.get_channel_member(channel.name, new_member.username)
+
+    auth.assert_add_channel_member(channel.name, new_member.role)
+
     if channel_member:
+        channel_member.role = new_member.role
+        db.commit()
+    else:
+        dao.create_channel_member(channel.name, new_member)
+
+
+@api_router.delete("/channels/{channel_name}/members", tags=["channels"])
+def delete_channel_member(
+    username: str,
+    channel: db_models.Channel = Depends(get_channel_or_fail),
+    dao: Dao = Depends(get_dao),
+    db=Depends(get_db),
+    auth: authorization.Rules = Depends(get_rules),
+):
+
+    auth.assert_list_channel_members(channel.name)
+    channel_member = dao.get_channel_member(channel.name, username)
+
+    if not channel_member:
         raise HTTPException(
-            status_code=status.HTTP_409_CONFLICT,
-            detail=f"Member {new_member.username} in {channel.name} exists",
+            status_code=status.HTTP_404_NOT_FOUND,
+            detail=f"channel member {username}/{channel.name} not found",
```

Comment:
Maybe it would be clearer to write "User {username} does not exist or is not a member of {channel.name}" or something along those lines?

Reply:
Sure! I'll go ahead and adjust it.

Reply:
should be good now